syntax = "proto3";

import "Velocities.proto";
message AirplaneLinearLocalPlannerStatus {
	double heading_target = 1;
}

message LinearLocalPlannerStatus {
	AirplaneLinearLocalPlannerStatus airplane_status = 1;
	uint32 current_path_section = 2;
	VelocityBody velocity_target = 3;
	double yaw_rate_target = 4;
	bool is_in_approach = 5;
}

message ManeuverLocalPlannerStatus{
	uint32 current_path_section = 1;
	double velocity_target = 2;
	double yaw_rate_target = 3;
	double climb_angle_target = 4;
	double heading_target = 5;
	bool is_in_approach = 6;
}

message LocalPlannerStatus {
	LinearLocalPlannerStatus linear_status = 1;
	ManeuverLocalPlannerStatus maneuver_status = 2;
}

message AirplaneLinearLocalPlannerParams {
	double k_altitude = 1;
	double k_heading = 2;
	double k_yaw_rate = 3;
}

message LinearLocalPlannerParams {
	AirplaneLinearLocalPlannerParams airplane_params = 1;
}

message ManeuverLocalPlannerParams {
	double k_altitude = 1;
	double k_convergence = 2;
	double k_yaw_rate = 3;
}

message LocalPlannerParams {
	LinearLocalPlannerParams linear_params = 1;
	ManeuverLocalPlannerParams maneuver_params = 2;
}